Enhancement: Warn about sync handler functions without an explicit `sync_to_thread` value (#1655)
litestar-org/litestar: docs/topics/sync-vs-async.rst

Sync vs. Async
==============

Litestar supports synchronous as well as asynchronous callables in almost all places
where it's possible to do so. In general, three different modes of execution are
supported:

- Running asynchronous callables directly
- Running synchronous callables directly
- Running synchronous callables in a thread pool

> There are a few important differences between those, which the following article tries
> to give an overview of.

Reviewer: ```suggestion
This article gives an overview of important differences between these modes.
```

Blocking and non-blocking
-------------------------

In the context of asynchronous programming, the terms *blocking* and *non-blocking*
are often used to describe a particular quality of a function: Blocking the flow of
execution.

Asynchronous functions are not inherently non-blocking. What they do instead is allow a
programmer to control exactly *where* they unblock and return control to the main loop,
allowing other asynchronous tasks to run, which is usually indicated by the use of the
``await`` keyword.  This is a very important aspect to note, since an async function
that never calls ``await`` and, for example, performs a computationally intensive task,
*will* block the main thread for its entire runtime, just like a synchronous function
would. More importantly, anything that happens inside an asynchronous function
*between* the parts where it waits will also be blocked.

Technically speaking, this means that there are no non-blocking functions in Python,
since async functions only "unblock" at each ``await``. As this is not a useful
definition of the term, "blocking" usually refers to callables that *block for a long

> time*. long period of time*.

Reviewer: ```suggestion
time*.
```
